modules/orchestrator/orchestrator-service/src/main/java/org/apache/airavata/orchestrator/server/OrchestratorServerHandler.java

```diff
@@ -146,10 +148,20 @@ public boolean launchExperiment(String experimentId, String gatewayId) throws TE
                 return false;
             }
 
+			UserConfigurationDataModel userConfigurationData = experiment.getUserConfigurationData();
             ComputeResourcePreference computeResourcePreference = registryClient.getGatewayComputeResourcePreference
 							(gatewayId,
-							experiment.getUserConfigurationData().getComputationalResourceScheduling().getResourceHostId());
+							userConfigurationData.getComputationalResourceScheduling().getResourceHostId());
             String token = computeResourcePreference.getResourceSpecificCredentialStoreToken();
+
+			if (userConfigurationData.getGroupResourceProfileId() != null) {
+				GroupComputeResourcePreference groupComputeResourcePreference = registryClient.getGroupComputeResourcePreference(
+						userConfigurationData.getComputationalResourceScheduling().getResourceHostId(),
+						userConfigurationData.getGroupResourceProfileId());
+				if (groupComputeResourcePreference.getResourceSpecificCredentialStoreToken() != null) {
+					token = groupComputeResourcePreference.getResourceSpecificCredentialStoreToken();
+				}
```

Comment:
If there is no resource specific credential store token on the groupComputeResourcePreference, it should then take the resource specific credential store token from the gateway computeResourcePreference.

Reply:
Also as we discussed, you're right about this one too. The code is correctly defaulting to the compute pref resource specific credential store token and overriding that if there is one defined for the GroupResourceProfile's compute pref.